pytest-embedded-idf/tests/test_idf.py

```diff
@@ -897,3 +897,96 @@ def test_erase_all_with_port_cache_case2(dut):
     )
 
     result.assert_outcomes(passed=2)
+
+
+def test_no_preserve_python_tests(testdir):
+    testdir.makepyfile(r"""
+        def test_python_case(dut):
+            dut.run_all_single_board_cases(name=["normal_case1", "multiple_stages_test"])
+    """)
+
+    testdir.runpytest(
+        '-s',
+        '--embedded-services', 'esp,idf',
+        '--app-path', os.path.join(testdir.tmpdir, 'unit_test_app_esp32'),
+        '--log-cli-level', 'DEBUG',
+        '--junitxml', 'report.xml',
+    )
+
+    junit_report = ET.parse('report.xml').getroot()[0]
+
+    assert junit_report.attrib['tests'] == '2'
+    for testcase in junit_report.findall('testcase'):
+        assert testcase.attrib['is_unity_case'] == '1'
+
+def test_preserve_python_tests(testdir):
+    testdir.makepyfile(r"""
+        def test_python_case(dut):
+            dut.run_all_single_board_cases(name=["normal_case1", "multiple_stages_test"])
+    """)
+
+    testdir.runpytest(
+        '-s',
+        '--embedded-services', 'esp,idf',
+        '--app-path', os.path.join(testdir.tmpdir, 'unit_test_app_esp32'),
+        '--log-cli-level', 'DEBUG',
+        '--junitxml', 'report.xml',
+        '--unity-test-report-mode',
+        'merge'
```

Comment:
```suggestion
        '--unity-test-report-mode', 'merge'
```

nitpick, keep the same format?